### DISCUSSION TITLE
iOS app build exceeds max build duration after upgrading to Xcode 15 build machine

### DISCUSSION THREAD

**Discussion by ghost** [codemagic-ci-cd/codemagic-docs#2079] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

Compile the app with Xcode 15 on Codemagic

### Expected results

It should compile and published

### Actual results

The build will hang on different part of the build process and eventually reach timeout even if `max_build_duration` is set to 120.

cases that have run into
- while compiling a storyboard
- Testing started (hang after this showed up in log) 

### Build id (optional)

65126eab558f7286dd6dee9e

**Comment by himesh-cm** [comment, marked as answer]
@mikechout ,

Just so you know, you should contact our team through the chat widget in the UI for a quicker response, since your team is on a paid plan.

It's difficult to debug the issue since the custom scripts you're running make it unclear which command is failing, and the logs are unhelpful.

Can you please connect with us via Chat, so we can take it forward?

**Reply by himesh-cm** [reply]
Hi @mikechout, Did you manage to solve it?

**Comment by ghost** [comment]
@himesh-cm I don't seem to find the chat widget button on the page, do i have to be the owner or admin of the team to see it ? thanks

**Reply by ghost** [reply]
I have changed different browser and still can't see the chat icon showing up at the bottom

**Reply by himesh-cm** [reply]
@mikechout, please disable adblocks, if any and try again.

**Reply by ghost** [reply]
I've reached out to the support team but the issue still exists, let me update after if resolved

**Comment by ollie-at-xdesign** [comment]
Hi @himesh-cm, we are also noticing build timeouts and 30min+ compile times when using the Xcode 15 boxes. Is there anyone that could help us investigate this? 

Thanks in advance!
Ollie

**Reply by himesh-cm** [reply]
Hi @ollie-at-xdesign, Please share a build id, so I can look into it.

**Reply by ollie-at-xdesign** [reply]
Hi @himesh-cm, the build id is 651acd3d5c6ae39f90011fd4.

Thank you for looking at this for us!

**Reply by himesh-cm** [reply]
Hi @ollie-at-xdesign,
A problem with Xcode 15 is causing simulators not to start up, which is a known [issue](https://developer.apple.com/forums/thread/735177).

The [issue](https://developer.apple.com/documentation/xcode-release-notes/xcode-15_1-release-notes) has been resolved with the release of Xcode 15.1 beta, but it's not currently available for use on Codemagic, as it was only released yesterday.

However, it will be included in the next base image update. I will also investigate whether older simulators can work with Xcode 15.

**Reply by ollie-at-xdesign** [reply]
Hi @himesh-cm, 

Thank you for investigating this and confirming the issue. 
We will continue to use XC 14 on our pipelines in the meantime until XC 15.1 is available. 

Cheers
OIlie
